frontend/src/javascript/utils/codeUtils.js

```diff
@@ -59,37 +60,82 @@ export function monacoChangeToLegacyFormat(code, change) {
     return result;
 }
 
```

Comment:
Dobrą praktyką jest ustawianie funkcji w kolejności od najważniejszych do najmniej ważnych. Zazwyczaj taki porządek jest równoważny porządkowi: od funkcji wysokiego poziomu do funkcji niskiego poziomu; a tym samym porządek: "Jeżeli funkcja A wywołuje funkcję B  to funkcja A powinna znajdować się wyżej niż funkcja B" (o ile tak się da, np. w programowaniu funkcyjnym w C++ musi być wręcz na odwrót).

Ma to kilka zalet, głównie to, że takie pliki czyta się trochę jak książkę, od góry do dołu. Jeżeli po raz pierwszy wchodzisz do jakiego pliku to na początku chcesz przeczytać te najważniejsze funkcje, żeby wgl zrozumieć za co dany plik jest odpowiedzialny, a potem dopiero interesuja Cię jakieś drobiazgi typu "w jaki sposób jest zaimplementowana funkcja isDigit". Co wiecej w przypadku tych drobnych funkcji jak `isLegalVariableCharacter` `isDigit`, `expandLeft`, `expandRight` po pierwszym zapoznaniu się z nimi ja już wiem co one rozwiązują i nie muszę do nich już wracać, dlatego lepiej, żeby były gdzieś na samym końcu pliku (a nie daj Boże jakby się te drobne funkcje mieszały kolejnością z tymi dużymi funkcjami, wtedy to by trzeba było skakać po całym pliku i za każdym razem natrafiać na te błahostki).

Dlatego zalecam zastosowanie tej praktyki w tym PR i w przyszłości :D